sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -38,8 +38,20 @@ enum class graph_state {
   executable, ///< In executable state, the graph is ready to execute.
 };
 
-// Forward declaration
-class node;
+/// Class representing a node in the graph, returned by command_graph::add().
+class __SYCL_EXPORT node {
+private:
+  node(const std::shared_ptr<detail::node_impl> &Impl) : impl(Impl) {}
+
+  template <class Obj>
+  friend decltype(Obj::impl)
+  sycl::detail::getSyclObjImpl(const Obj &SyclObject);
+  template <class T>
+  friend T sycl::detail::createSyclObjFromImpl(decltype(T::impl) ImplObj);
+
+  std::shared_ptr<detail::node_impl> impl;
```

Comment:
Also unrelated to this PR, but we might to consider renaming `impl` to something like `MImpl` to not violate naming conventions.

Reply:
This caught my eye too previously, but the helpers for getting/setting SYCL objects from impl rely on this naming convention https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/detail/common.hpp#L290-L304